mllib/src/main/scala/org/apache/spark/ml/feature/RFormula.scala

```diff
@@ -37,6 +37,42 @@ import org.apache.spark.sql.types._
  */
 private[feature] trait RFormulaBase extends HasFeaturesCol with HasLabelCol {
 
+  /**
+   * Param for how to order categories of a string FEATURE column used by `StringIndexer`.
+   * The last category after ordering is dropped when encoding strings.
+   * Supported options: 'frequencyDesc', 'frequencyAsc', 'alphabetDesc', 'alphabetAsc'.
+   * The default value is 'frequencyDesc'. When the ordering is set to 'alphabetDesc', `RFormula`
+   * drops the same category as R when encoding strings.
+   *
+   * The options are explained using an example `'b', 'a', 'b', 'a', 'c', 'b'`:
+   * {{{
+   * +-----------------+---------------------------------------+----------------------------------+
+   * |      Option     | Category mapped to 0 by StringIndexer |  Category dropped by RFormula    |
+   * +-----------------+---------------------------------------+----------------------------------+
+   * | 'frequencyDesc' | most frequent category ('b')          | least frequent category ('c')    |
+   * | 'frequencyAsc'  | least frequent category ('c')         | most frequent category ('b')     |
+   * | 'alphabetDesc'  | first alphabetical category ('a')     | last alphabetical category ('c') |
+   * | 'alphabetAsc'   | last alphabetical category ('c')      | first alphabetical category ('a')|
```

Comment:
I think ```ascending``` is going up i.e. A-Z, ```descending``` is going down i.e. Z-A. So in this example, first alphabetical category is "c" if order by ```alphabetDesc```, first alphabetical category is "a" if order by ```alphabetAsc```. Please correct me if I have misunderstand.